synapse/util/retryutils.py

```diff
@@ -35,6 +35,29 @@ def __init__(self, retry_last_ts, retry_interval, destination):
 
 
 @defer.inlineCallbacks
+def would_retry(destination, clock, store):
+    """Check if ``get_retry_limiter`` would throw. This is purely to allow
+    things to bail early if the resulting request would not otherwise be sent
+    """
```

Comment:
please can we document types of args and return values, even if it's "obvious". I'm fed up with figuring it all out from first principles in the synapse codebase. You can even c&p most of it from `get_retry_limiter` in this case.